src/Compiler/CodeGen/IlxGen.fs

```diff
@@ -10692,7 +10696,15 @@ and GenTypeDef cenv mgbuf lazyInitInfo eenv m (tycon: Tycon) =
                             let ilPropName = fspec.LogicalName
                             let ilMethName = "get_" + ilPropName
                             let access = ComputeMemberAccess isPropHidden
-                            yield mkLdfldMethodDef (ilMethName, access, isStatic, ilThisTy, ilFieldName, ilPropType)
+                            let isStruct = tcref.IsFSharpStructOrEnumTycon && not tcref.IsEnumTycon
```

Comment:
Just curious, why the check is implemented this way? 
At least in the IL code something else seems to be [analyzed](https://github.com/dotnet/fsharp/blob/8882026ada8f038fd2610d8823e7044bead247bf/src/Compiler/AbstractIL/il.fs#L2693).

Reply:
Seemed like the easiest way to do it here. I'm definitely open to other suggestions. Maybe we should add `IsFShartStructTycon`. Or maybe just use `IsFSharpStructOrEnumTycon` because `enum` will never reach this place.

Reply:
Leaving it up to you, just voting for consistent approach if applicable

Reply:
I don't think we can do it like in `ILTypeDef`, but I did find [another way](https://github.com/dotnet/fsharp/blob/8882026ada8f038fd2610d8823e7044bead247bf/src/Compiler/TypedTree/TypedTreeOps.fs#L1957) which might be better.

Reply:
Anyway, I'm definitely okay with the current approach